SECEdgar/filings/cik_validator.py

```diff
@@ -0,0 +1,100 @@
+import warnings
+
+from SECEdgar.base import _EDGARBase
+
+from SECEdgar.utils.exceptions import CIKError, EDGARQueryError
+
+
+class CIKValidator(_EDGARBase):
+    """Validates company tickers and/or company names based on CIK availability.
+
+    Used internally by the CIK class. Not intended for outside use.
+
+    Args:
+        lookups (Union[str, list, tuple]): List of tickers and/or company names for
+            which to find CIKs.
+        **kwargs: Any keyword arguments needed to be passed to
+            _EDGARBase (see class for more details).
+
+    .. versionadded:: 0.1.5
+    """
+
+    def __init__(self, lookups, **kwargs):
+        super(CIKValidator, self).__init__(**kwargs)
+        if isinstance(lookups, str):
+            self._lookups = [lookups]
+        else:
+            try:
+                if not all(type(o) is str for o in lookups):
+                    raise TypeError("CIKs must be given as string or iterable.")
+                self._lookups = lookups
+            except TypeError:
+                raise TypeError("CIKs must be given as string or iterable.")
+        self._params['action'] = 'getcompany'
+
+    @property
+    def url(self):
+        return "cgi-bin/browse-edgar"
```

Comment:
better name would be `resource`.

Reply:
I looked it up. Looks like the official name is "path". See link [here](https://community.tealiumiq.com/t5/iQ-Tag-Management/URL-Components-Explained/ta-p/5573#toc-hId-678478843).